common/flatpak-installation.c

```diff
@@ -1093,6 +1167,47 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
       if (remote_commit != NULL &&
           g_strcmp0 (remote_commit, local_commit) != 0)
         g_ptr_array_add (updates, g_object_ref (installed_ref));
+      else if (g_str_has_prefix (full_ref, "app/"))
+        {
+          g_autoptr(GVariant) deploy_data = NULL;
+          const gchar *runtime = NULL;
+          g_autofree gchar *full_runtime_ref = NULL;
+          g_autoptr (GError) local_error = NULL;
+
+          /* Check if all "should-download" related refs for an installed app are installed.
+           * If not, mark that app's installed ref as updatable again so that it
+           * can be auto-installed via FlatpakTransaction's update API.
+           * This makes sure that app remains in "usable" state and fixes itself
+           * through an update.
+           */
+          if (flatpak_installation_check_installed_ref_missing_related_ref (self,
+                                                                            remote_name,
+                                                                            full_ref,
+                                                                            cancellable))
```

Comment:
You'll want to check `flatpak_dir_ref_is_masked` before adding things to `updates`

Reply:
There is a check above in the loop for the main installed ref. Or do you mean `flatpak_dir_ref_is_masked` check for extension refs?

Reply:
Yes: if an extension is missing because it is masked, apps that would use it should not be marked uodate-able on its accounf